mne/io/meas_info.py

```diff
@@ -1303,6 +1303,10 @@ def write_meas_info(fid, info, data_type=None, reset_range=True):
     # Measurement info
     start_block(fid, FIFF.FIFFB_MEAS_INFO)
 
+    # Add measurement id
+    if info['meas_id'] is not None:
+        write_id(fid, FIFF.FIFF_PARENT_BLOCK_ID, info['meas_id'])
+
```

Comment:
Is this check needed or wanted?

https://github.com/mne-tools/mne-python/blob/ab4571a83c4e5d5de0dcb2e9346f809f113dd619/mne/io/write.py#L224-L226
will put in the default (DATE_NONE)

I guess the question is do you always want a meas_id in a the fiff structure?

Reply:
The FIF file itself will need one, but I don't know if `meas_info` needs one. Can you try writing a file with an `info` without `meas_id` in it and see if the C tools can read it?

Reply:
I don't have the C tools running or any experience with them. 

For what its worth without this line (ie the `info['meas_id'] == None` case) it would revert to the backend of write_info method so if those played nice with the C code I imagine this would be fine as well

In those cases (ie the `info['meas_id'] == None`) read_info in mne-python returned an info['meas_id'] with the DATE_NONE usecs in it.

Also the file gets the default (None/DATE_NONE) id.